homeassistant/components/intellifire/coordinator.py

```diff
@@ -0,0 +1,60 @@
+"""The IntelliFire integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+import logging.handlers
+
+from async_timeout import timeout
+from intellifire4py import IntellifireAsync, IntellifirePollData
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER
+
+
+class IntellifireDataUpdateCoordinator(DataUpdateCoordinator[IntellifirePollData]):
+    """Class to manage the polling of the fireplace API."""
+
+    def __init__(self, hass: HomeAssistant, api: IntellifireAsync) -> None:
+        """Initialize the Coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=15),
+            update_method=self._async_update_data,
+        )
+        self._api = api
+
+    async def _async_update_data(self):
+        LOGGER.debug("Calling update loop on IntelliFire")
+        async with timeout(100):
+            try:
+                await self._api.poll(logging_level=logging.DEBUG)
+            except Exception as exception:
+                raise UpdateFailed from exception
+        return self._api.data
+
+    @property
+    def serial(self) -> str:
```

Comment:
Personally I would put the serial and device_info property in a seperate `IntellifireEntity` class and just read the value from `self.coordinator.data.serial`. 

Don't think this should be a blocker for the PR, but others should confirm. 😉

Reply:
Well as a matter of fact, due to other (processed) review comments, this method became unused 😉